src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,248 @@
+	 * @return imageMap - The HashMap which stores the userName and userImage.
+	 */
+	protected static HashMap<String, String> getImageMap() {
+		logMessage = "getImageMap method of ContentCommunicator class is executing";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		return imageMap;
+	}
+	
+	/**
+	 * This method allows other classes within this package to set the HashMap imageMap.
+	 * @param imagemap - The HashMap which stores the userName and userImage of all clients.
+	 */
+	protected static void setImageMap(HashMap<String, String> imagemap) {
+		logMessage = "setImageMap method of ContentCommunicator class is executing";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		imageMap = imagemap;
+	}
+	
+	/**
+	 * This method provides HashMap handlerMap to other classes within this package.
+	 * @return handlerMap - The HashMap which stores the identifier and handler of subscriber of this module.
+	 */
+	protected static HashMap<String, IContentNotificationHandler> getHandlerMap() {
+		logMessage = "getHandlerMap method of ContentCommunicator class is executing";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		return handlerMap;
+	}
+	
+	/**
+	 * This method initializes a user by storing its username and userimage in a map as Strings.
+	 * This method stores these values locally too and generates server's ipAddress by using port.
+	 * Finally, some manipulation on JSON string is done and then sent to server via networking.
+	 * @param userDetails - This is a JSON string which contains a JSON object with three keys, 
+	 * namely ipAddress, username and image.
+	 */
 	@Override
 	public void initialiseUser(String userDetails) {
-    	
-    }
+		logMessage = "initialiseUser method of ContentCommunicator class is executing";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		JSONObject jsonObject;
+		try {
+			jsonObject = new JSONObject(userDetails);
+		} catch(Exception e) {
+			logMessage = "Argument userDetails cannot be converted to JSON object";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		try {
+			serverIPAddress = jsonObject.getString("ipAddress");
+		} catch(Exception e) {
+			logMessage = "the value of ipAddress key in argument userDetails is not a String";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		try {
+			userName = jsonObject.getString("username");
+		} catch(Exception e) {
+			logMessage = "the value of username key in argument userDetails is not a String";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		try {
+			userImage = jsonObject.getString("image");
+		} catch(Exception e) {
+			logMessage = "the value of image key in argument userDetails is not a String";
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		serverIPAddress += ":";
+		serverIPAddress += String.valueOf(port);

```

Comment:
port value is not being obtained again. Currently just taking in the default value. There needs to be a check here

Reply:
ok, I will get the port value again here.

Reply:
port value is obtained here only now. An additional check is added on port value. And 0 is passed while initialising networking communicator.

Reply:
A check for port value = zero is fine but you need handle that case. Just returning from the program will render us unable to send any message and eventually the program will crash. You need to wait till you get the correct port(Checking in every iteration of the loop) and then compile the correct ipAddress

Reply:
Does Thread.sleep() need any imports?

Reply:
No